### PR TITLE
7. Vote on a review

### PR DESCRIPTION
**Pull request by roisinoreilly**

### CODE REVIEW

File: src/components/Votes.jsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    }
+  };
+
+  const downvote = () => {
+    if (hasVotedDown === false) {
+    setUpvotes((currentVotes) => currentVotes - 1);
+    increaseVotes(review.review_id, -1);
+    setHasVotedDown(true)
+    }
+  };
```
Comment (douglashellowell) [inline src/components/Votes.jsx]: these functions look very similar to me 🔍

could you refactor them into one?

File: src/components/Votes.jsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  return (
+    <div>
+      <button onClick={upvote}>upvote</button>
+      <button onClick={downvote}>downvote</button>
+      <div>Votes: {upvotes}</div>
```
Comment (douglashellowell) [inline src/components/Votes.jsx]: `<div>` tags are great for grouping and stlying but you should use different tags for just text!

**Comment by douglashellowell** [comment]
Great work but there are a couple of odd behaviours you'll need to sort out before you merge

---

Currently you can upvote and downvote once but:

- There's no indication which button(s) can /  can't be clicked. Have a look at the `disabled` attribute and/or change the stlying to indicate what the user can do
- Once you've clicked both buttons they're both locked until you refresh the page! It would be nice to have the option to change your mind

---

If the patch req fails or if a user loses connection the number still changes!
You need to indicate to the user that the vote failed somehow

---

Please put in another `/nchelp pr` when you've got these bits sorted please! 🚀

**Comment by dappernerddesigns** [comment]
Nice refactor on the api call.

As per Doug's feedback above, when the connection fails, a user can still change the number visually on screen. 

Can you show the user it hasn't worked when there is no connection? (You can mimic no connection in inspect tools in browser). Drop in another PR when you have that bit squared away.

**Review by dappernerddesigns — APPROVED** [review]
Really nicely handled. I like the disabled buttons and the error message to the user. Top stuff!! Merge away.

**Pull request merged**